src/OrbitGl/CaptureViewElement.h

```diff
@@ -17,9 +18,9 @@ class GlCanvas;
 namespace orbit_gl {
 
 /* Base class for UI elements drawn underneath the capture window. */
-class CaptureViewElement : public Pickable {
+class CaptureViewElement : public Pickable, public AccessibleInterfaceProvider {
  public:
-  explicit CaptureViewElement(CaptureViewElement* parent, TimeGraph* time_graph,
+  explicit CaptureViewElement(AccessibleInterfaceProvider* parent, TimeGraph* time_graph,
```

Comment:
As discussed offline: I would prefer to have the parent / children of our Composite pattern to be of the same type, so children can have access to the parent's position, size etc.

So I would leave this as `CaptureViewElement*`, and keep the special case in `TimeGraph` - but replace `GetCanvas()` with a new method that returns an `AccessibleInterfaceProvider`.

Reply:
Hmm, but maybe then we should simply rename `AccessibleInterfaceProvider` to something more general that has a position and size. GlCanvas being a sort of `CaptureViewElement` would be the cleanest solution, but `CaptureViewElement` already contains more than just pos, size, parent and accessibility. Essentially, GlCanvas is the parent of TimeGraph, so GetParent should be used.

Reply:
IMO we shouldn't have this driven too much by the accessibility. The way it's build right now, `TimeGraph` for me is the clear root of the hierarchy of rendered elements, not `GlCanvas`. Having TimeGraph return `nullptr` as parent is logical in that sense.

I've always thought of it rather this way: How would we do it if accessibility wasn't involved? That should be the clean way. The accessibility needs to bridge that gap from GlCanvas somehow, and it's already a bit of a complex construct because we jump from the Qt world to Gl, and I would like to have this special cases hidden in the accessibility classes instead of the control hierarchy.

I already said this offline: I do not have a strong veto against this solution, and everything except the change in type for the parent makes sense to me, but I feel that the type change will cause issues in the future and does not reflect the meaning of "parent" because it is too prominently driven by accessibility.

Reply:
To keep it short: I also don't have a strong opinion in this, so I'll let Pablo decide.

I think you could argue that GlCanvas could be considered being a parent of TimeGraph even completely without accessibility (where I totally agree with you that accessibility impl. details should not drive our design), esp. with respect to pos/size.

Reply:
I feel that ideally CaptureWindow might be the root of the OrbitGl tree. It depends on how we want to divide different TimeGraphs. But for sure, the type of all the elements in this case (CaptureWindows included) should be CaptureViewElement. So, for now I will keep this (CaptureViewElement as type for parents, nullptr as the father of TimeGraph) by having a different type for TimeGraph's parent, until we reach an state of several TimeGraph, when we should take the decision about which should be the root.